### PR TITLE
Select server view : Fix next button action

### PR DESCRIPTION
**Pull request by dhekra-rouatbi**
For more information on contributing to Tella source code, see the [Contributor Guidelines](contributing/contributor_guide.md).

## Type of change

**Description:**

### CODE REVIEW

File: Tella/Scenes/Settings/Views/Servers/ServerSelectionView.swift [hunk 1]
```diff
@@ -95,14 +94,14 @@ struct ServerSelectionView: View {
     }
     
     fileprivate func navigateToGDriveFlow() {
-        serversViewModel.shouldHideNextButton = true
+        serversViewModel.shouldEnableNextButton = false
```
Comment (ValbuenaG) [inline Tella/Scenes/Settings/Views/Servers/ServerSelectionView.swift:97]: could we add here:

`serversViewModel.selectedServerType = nil`

If the user cancel the sign in flow the drive connection would be kept as selected and the next button disabled

Comment (ValbuenaG) [reply Tella/Scenes/Settings/Views/Servers/ServerSelectionView.swift:97]: Screenshot 2024-08-28 at 2 24 14 PM

Comment (dhekra-rouatbi) [reply Tella/Scenes/Settings/Views/Servers/ServerSelectionView.swift:97]: yeah you're right, I have the same issue
What do you think if we add  `serversViewModel.shouldEnableNextButton = true` to 
```
.onReceive(gDriveVM.$signInState){ signInState in
            if case .error(let message) = signInState {
                serversViewModel.shouldEnableNextButton = true
                Toast.displayToast(message: message)
            }
        }
        

```

Comment (dhekra-rouatbi) [reply Tella/Scenes/Settings/Views/Servers/ServerSelectionView.swift:97]: I'll try also to add `serversViewModel.selectedServerType = nil`

Comment (dhekra-rouatbi) [reply Tella/Scenes/Settings/Views/Servers/ServerSelectionView.swift]: I fixed the issue by adding a `resetView`  function  
```
fileprivate func resetView() {
        serversViewModel.selectedServerType = nil
        serversViewModel.shouldEnableNextButton = false
    }

```

**Review by ValbuenaG — APPROVED** [review]

**Pull request merged**